src/test/java/org/s1ck/gdl/predicates/booleans/AndTest.java

```diff
@@ -20,4 +26,27 @@ public void convertToCnfTest() {
 
     assertEquals(reference,and.toCNF());
   }
+
+  @Test
+  public void extractVariablesTest() {
+    Comparison a = new Comparison(
+            new ElementSelector("a"),
+            Comparison.Comparator.EQ,
+            new ElementSelector("b")
+    );
+
+    Comparison b = new Comparison(
+            new PropertySelector("a","label"),
+            Comparison.Comparator.EQ,
+            new Literal("Person")
+    );
+
+    And and = new And(a,b);
+
+    Set<String> reference = new HashSet<>();
+    reference.add("a");
+    reference.add("b");
+
+    assertEquals(reference,and.variables());
```

Comment:
additionally test the not equals case (e.g. after adding "a" to the set and adding "c" after "b")